odbc/src/tests.rs

```diff
@@ -463,3 +465,401 @@ mod get_diag_rec {
+        let attr_buffer = Box::into_raw(Box::new(0));
+        let string_length_ptr = &mut 0;
+        assert_eq!(
+            SqlReturn::SUCCESS,
+            SQLGetEnvAttrW(
+                env_handle as *mut _,
+                EnvironmentAttribute::ConnectionPooling,
+                attr_buffer as Pointer,
+                5,
+                string_length_ptr
+            )
+        );
+        assert_eq!(ConnectionPooling::Off, unsafe {
+            *(attr_buffer as *const ConnectionPooling)
+        });
+        assert_eq!(4, *string_length_ptr);
+
+        // SQLSetEnvAttrW(SQL_ATTR_CONNECTION_POOLING, SQL_CP_DRIVER_AWARE)
+        assert_eq!(
+            SqlReturn::SUCCESS_WITH_INFO,
+            SQLSetEnvAttrW(
+                env_handle as HEnv,
+                EnvironmentAttribute::ConnectionPooling,
+                Pointer::from(AttrConnectionPooling::DriverAware),
+                0
+            )
+        );
+
+        // SQLGetEnvAttrW(SQL_ATTR_CONNECTION_POOLING)
+        assert_eq!(
+            SqlReturn::SUCCESS,
+            SQLGetEnvAttrW(
+                env_handle as *mut _,
+                EnvironmentAttribute::ConnectionPooling,
+                attr_buffer as Pointer,
+                5,
+                string_length_ptr
+            )
+        );
+        assert_eq!(ConnectionPooling::Off, unsafe {
+            *(attr_buffer as *const ConnectionPooling)
+        });
+        assert_eq!(4, *string_length_ptr);
+        unsafe { Box::from_raw(attr_buffer) };
+
+        // SQLSetEnvAttrW(SQL_ATTR_CONNECTION_POOLING, SQL_CP_ONE_PER_HENV)
+        assert_eq!(
+            SqlReturn::SUCCESS_WITH_INFO,
+            SQLSetEnvAttrW(
+                env_handle as HEnv,
+                EnvironmentAttribute::ConnectionPooling,
+                Pointer::from(AttrConnectionPooling::OnePerHenv),
+                0
+            )
+        );
+
+        // Initialize buffers
+        let sql_state = &mut [0u16; 6] as *mut _;
+        let message_text = &mut [0u16; 71] as *mut _;
+        let text_length_ptr = &mut 0;
+        let native_err_ptr = &mut 0;
+        assert_eq!(
+            SqlReturn::SUCCESS_WITH_INFO,
+            SQLGetDiagRecW(
+                HandleType::Env,
+                env_handle as *mut _,
+                1,
+                sql_state,
+                native_err_ptr,
+                message_text,
+                71,
+                text_length_ptr
+            )
+        );
+        assert_eq!(OPTIONAL_VALUE_CHANGED, unsafe {
+            String::from_utf16(&*(sql_state as *const [u16; 6])).unwrap()
+        });
+        assert_eq!(
+            "[MongoDB][API] Invalid value for attribute SQL_ATTR_CONNECTION_POOLING\0",
+            unsafe { String::from_utf16(&*(message_text as *const [u16; 71])).unwrap() }
+        );
+        // Exclude the number of characters required for the null terminator
+        assert_eq!(70, *text_length_ptr);
```

Comment:
Keeping the assert on the error code and error message is great because we want to make sure that we get the "right error, but I don't think you need to add these 2 asserts. You already added tests for SQLGetDiagRec and they cover this.